### PR TITLE
Only lint if @flow pragma is present

### PR DESCRIPTION
**Pull request by nsfmc**
doesn't intelligently check for it in the first comment like flow will,
but this check seems to be good enough to halt spurious flow checks.

additionally, uses the view's contents piped to stdin to check for
errors rather than using a tempfile.

real talk: not sure if there's any harm in switching from tempfile to this, but i do know that the flow pragma part is not entirely bad, so it's maybe a wash. happy to update the pr in any event.

hope alls well,
--marcos

### CODE REVIEW

File: linter.py [hunk 1]
```diff
@@ -35,6 +34,20 @@ class Flow(Linter):
 
     __flow_near_re = '`(?P<near>[^`]+)`'
 
+    def run(self, cmd, code):
+        """
+        flow lint code if `@flow` pragma is present
```
Comment (ckaznocha) [inline linter.py]: It silly but would you mind adding a period after `present` so that the liter passes?

**Comment by ckaznocha** [comment]
Hey @nsfmc Thanks for the PR! The check for the flow comment looks great and I'd accept it right away but I want to spend a little time testing the switch from using temp file. Would it be to much trouble for you to split that into a separate PR?

File: linter.py (hunk 1 above)
Comment (nsfmc) [inline linter.py]: no problem at all

**Comment by nsfmc** [comment]
sure thing @ckaznocha , i've been using it like this for a few weeks already, but i'll factor out the `@flow` pragma stuff into a separate pr from the tempfile stuff. it's very similar, but the two are slightly entangled at the moment.

**Comment by nsfmc** [comment]
testing this again after rebasing onto the most recent master (sorry i've been crazy busy these past few months), will report back if this is not totally busted like it was before

**Comment by nsfmc** [comment]
just leaving a comment here because this issue discussion reminded me to update this branch https://github.com/SublimeLinter/SublimeLinter-flow/issues/41#issuecomment-252828146 

this currently works with flow 0.32+, i'm not sure when the json output solidified, but when the actual flow output got fancy, i think it probably came along for the ride. I'm not totally sold on how i'm handling some of these errors (flow issues can be complex) but it has been usable for me for the past few weeks or so. again, any feedback/comments would help. 

hope alls well.

File: linter.py [hunk 2]
```diff
@@ -46,50 +59,95 @@ def cmd(self):
...
+        line = message_start.get('line', None)
+        if line:
+            line -= 1
+        col = message_start.get('column', None)
+        if col:
+            col -= 1
+        end = message_end.get('column', None)
+
+        # slice the error message from the context and loc positions
+        if end and line == (message_end.get('line') - 1):
```
Comment (nsfmc) [inline linter.py]: for multiline errors, i punt and don't even try to match the text, instead, i let sublimelinter highlight the first character of the error and hope that the context messages are useful enough (they're usually ok)

File: linter.py [hunk 3]
```diff
@@ -46,50 +59,95 @@ def cmd(self):
...
+            error_context = error_context[col:end]
+            persist.debug('`{}`'.format(error_context))
```
Comment (nsfmc) [inline linter.py]: i should remove this debug line, whoops

File: linter.py [hunk 4]
```diff
@@ -46,50 +59,95 @@ def cmd(self):
...
+        else:
+            error_context = None
+
+        persist.debug(error_context)
```
Comment (nsfmc) [inline linter.py]: and this one

File: linter.py (hunk 3 above)
Comment (ckaznocha) [inline linter.py]: Everything else looks good to me. Want to take out the debut statements then I'll merge

**Comment by nsfmc** [comment]
@ckaznocha pushed the branch with changes a while ago and ran a very ad-hoc test against a stockpile of flow errors from my workplace repo. at the very minimum the lint command no longer dies on some forms of input and seems to give sensible composed error messages for most of them.

**Pull request merged**